ddtrace/contrib/jinja2/patch.py

```diff
@@ -0,0 +1,131 @@
+    if cfg['inherit_service']:
+        if span._parent and span._parent.service:
+            return span._parent.service
+    return cfg['service_name']
+
+
+def _wrap_generate(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(instance.environment)
+    if not pin or not pin.enabled():
+        for s in wrapped(*args, **kwargs):
+            yield s
+
+    template_name = instance.name or DEFAULT_TEMPLATE_NAME
+    with pin.tracer.trace("jinja2.generate", span_type=http.TEMPLATE) as span:
+        # update the span service name before doing any action
+        span.service = _get_service_name(instance.environment, span)
+        try:
+            for s in wrapped(*args, **kwargs):
+                yield s
+        finally:
+            span.resource = template_name
+            span.set_tag("jinja2.template_name", template_name)
+
+
+def _wrap_render(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(instance.environment)
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    template_name = instance.name or DEFAULT_TEMPLATE_NAME
+    print("render tracer", id(pin.tracer))
+    with pin.tracer.trace("jinja2.render", span_type=http.TEMPLATE) as span:
+        # update the span service name before doing any action
+        span.service = _get_service_name(instance.environment, span)
+        try:
+            return wrapped(*args, **kwargs)
+        finally:
+            span.resource = template_name
+            span.set_tag("jinja2.template_name", template_name)
+
+
+def _wrap_compile(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    if len(args) > 1:
+        template_name = args[1]
+    else:
+        template_name = kwargs.get("name", DEFAULT_TEMPLATE_NAME)
+
+    with pin.tracer.trace("jinja2.compile", span_type=http.TEMPLATE) as span:
+        # update the span service name before doing any action
+        span.service = _get_service_name(instance, span)
+        try:
+            return wrapped(*args, **kwargs)
+        finally:
+            span.resource = template_name
+            span.set_tag("jinja2.template_name", template_name)
+
+
+def _wrap_load_template(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    template_name = kwargs.get("name", args[0])
+    with pin.tracer.trace("jinja2.load", span_type=http.TEMPLATE) as span:
+        # update the span service name before doing any action
+        span.service = _get_service_name(instance, span)
+        template = None
+        try:
+            template = wrapped(*args, **kwargs)
+            return template
+        finally:
+            span.resource = template_name
+            span.set_tag("jinja2.template_name", template_name)
+            if template:
+                span.set_tag("jinja2.template_path", template.filename)
```

Comment:
If we used a different loader will `template.filename` not exist?

Reply:
`filename` is set in `Template` (`_from_namespace`), So it seems fine to me

Reply:
cool. sorry for not checking there myself :(